src/test/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```diff
@@ -0,0 +1,566 @@
+     *
+     * @param ast javadoc ast
+     */
+    public void scrapeContent(DetailNode ast) {
+        if (ast.getType() == JavadocTokenTypes.PARAGRAPH) {
+            if (isParentText(ast)) {
+                currentStatus = ScrapeStatus.PARENT;
+                parentSectionStartIdx = getParentIndexOf(ast);
+                moduleDetails.setParent(getParentText(ast));
+            }
+            else if (isViolationMessagesText(ast)) {
+                currentStatus = ScrapeStatus.VIOLATION_MESSAGES;
+            }
+            else if (exampleSectionStartIdx == -1
+                    && isExamplesText(ast)) {
+                exampleSectionStartIdx = getParentIndexOf(ast);
+            }
+        }
+        else if (ast.getType() == JavadocTokenTypes.LI) {
+            if (isPropertyList(ast)) {
+                currentStatus = ScrapeStatus.PROPERTY;
+
+                if (propertySectionStartIdx == -1) {
+                    propertySectionStartIdx = getParentIndexOf(ast);
+                }
+
+                moduleDetails.setDescription(descriptionText);
+                moduleDetails.addToProperties(createProperties(ast));
+            }
+            else if (currentStatus == ScrapeStatus.VIOLATION_MESSAGES) {
+                moduleDetails.addToViolationMessages(getViolationMessages(ast));
+            }
+        }
+    }
+
+    /**
+     * Create the modulePropertyDetails content.
+     *
+     * @param nodeLi list item javadoc node
+     * @return modulePropertyDetail object for the corresponding property
+     */
+    private static ModulePropertyDetails createProperties(DetailNode nodeLi) {
+        final ModulePropertyDetails modulePropertyDetails = new ModulePropertyDetails();
+        final DetailNode propertyNameTag =
+                getFirstChildOfType(nodeLi, JavadocTokenTypes.JAVADOC_INLINE_TAG, 0);
+        final DetailNode propertyType = getFirstChildOfMatchingText(nodeLi, TYPE_TAG);
+
+        final String propertyDesc = DESC_CLEAN.matcher(
+                constructSubTreeText(nodeLi, propertyNameTag.getIndex() + 1,
+                propertyType.getIndex() - 1))
+                .replaceAll(Matcher.quoteReplacement(""));
+
+        modulePropertyDetails.setDescription(propertyDesc.trim());
+        modulePropertyDetails.setName(getTextFromTag(propertyNameTag));
+        modulePropertyDetails.setType(
+                getTagTextFromProperty(nodeLi,
+                getFirstChildOfMatchingText(nodeLi, TYPE_TAG)));
+        final DetailNode validationTypeNode = getFirstChildOfMatchingText(nodeLi,
+                VALIDATION_TYPE_TAG);
+        if (validationTypeNode != null) {
+            modulePropertyDetails.setValidationType(getTagTextFromProperty(nodeLi,
+                    validationTypeNode));
+        }
+        final String defaultValue = getPropertyDefaultText(nodeLi);
+        if (!"null".equals(defaultValue)
+            && !"the charset property of the parent".equals(defaultValue)) {
```

Comment:
Please extract this special default values into list and change this to List#contains

Reply:
Done